core/dbt/tests/util.py

```diff
@@ -73,7 +73,11 @@ def run_dbt(args: List[str] = None, expect_pass=True):
         args = ["run"]
 
     print("\n\nInvoking dbt with {}".format(args))
-    res, success = handle_and_check(args)
+    # ctx = DBTContext.from_args(args)
+    ctx = DBTContext(args)
```

Comment:
This `ctx` object here is actually a context object for the actual command( for example `run`), and in the bottom we are calling the invoke of the top level command(`cli` group) with it. 

What actually happened during the process of calling a click framework from the cli is(lets use `run` as an example):
- invoke the `cli group cmd` with context built for that group(doesn't actually have anything related to run except that the `invoked_subcommand ` is set) [link](https://github.com/pallets/click/blob/main/src/click/core.py#L1653-L1654)
- invoke the run command with context that was built for run, with context that went through `cli group cmd` as parent. [link](https://github.com/pallets/click/blob/main/src/click/core.py#L1655-L1657)
- not sure what exactly happens if we have multi level group like `docs`

So this actually doesn't mimic what happens in cli, we probably need to figure out what is the best way to do this

Reply:
> This ctx object here is actually a context object for the actual command( for example run)

I believe the ctx object is for the top-level command, as constructed [here](https://github.com/dbt-labs/dbt-core/pull/6387/files#diff-6b14f723aa6f9a670e0994eea8dbddf5a973ef12c3c4a1e48461bad094c2de58R13). The subcommand context isn't used, just created for validation [here](https://github.com/dbt-labs/dbt-core/pull/6387/files#diff-6b14f723aa6f9a670e0994eea8dbddf5a973ef12c3c4a1e48461bad094c2de58R16).

Reply:
But maybe what you are doing here will mimic the behavior of calling in from https://github.com/pallets/click/blob/main/src/click/core.py#L1055 ?